### PR TITLE
fix: output image title attribute

### LINKED ISSUE #308: Image replacement title property is not output by the pipeline
**Description**

Alternative title is not output by the pipeline although it is set properly on an image located inside a word document. 

Alternative description is output properly as `alt` attribute on the `img` tag.

**To Reproduce**

1. Create a word document in Sharepoint
2. Add an image in the document
3. Set the alternative title to "I went through the pipeline"
4. Preview the document
5. Render the document as .md and find the title "I went through the pipeline" in the output
6. Render the document as HTML and see that the title is not in the output

If you can access it, you may also use [this word document](https://adobe.sharepoint.com/:w:/s/HelixProjects/ETNu8FsJu0ZHi_Ue9w89JzAB56Q-do2YrfSnFbiFvUqaUA?e=012ZN1)

**Expected behavior**

I expect the replacement title value set on an image within a word document to be output as `title` attribute on the `img` tag.

**Screenshots**

The following screenshot shows the alternative title and descriptions set on an image in within a word document

alternative

The following screenshot shows the Markdown output of the document that contains the image. The alternative title is present in the output, as expected.

md-fine

The following screenshot shows that the alternative title is not contained in the HTML output.

Screenshot 2023-05-05 at 21 12 37

**Version:**
14.21.2

**Additional context**

https://cq-dev.slack.com/archives/C04RN3CQWBZ/p1683295645721079?thread_ts=1683295547.817699&cid=C04RN3CQWBZ

### PR DESCRIPTION
**Pull request by tmaret**
fix #308

### CODE REVIEW

**Comment by tmaret** [comment]
Thanks @tripodsan ! I merged your suggestion commit.

**Comment by tmaret** [comment]
@tripodsan thanks for point this out, I have pushed commits that include the `data-title` attribute in test fixtures.

**Review by tripodsan — CHANGES_REQUESTED** [review]
avoid empty string attributes

File: test/fixtures/content/image-from-meta.html [hunk 1]
```diff
@@ -26,7 +26,7 @@ <h1 id="hero">Hero</h1>
                 <source type="image/webp" srcset="./media_a22b1a53edf9b324465d14b2efca169a25d564a0.png?width=2000&#x26;format=webply&#x26;optimize=medium" media="(min-width: 600px)">
                 <source type="image/webp" srcset="./media_a22b1a53edf9b324465d14b2efca169a25d564a0.png?width=750&#x26;format=webply&#x26;optimize=medium">
                 <source type="image/png" srcset="./media_a22b1a53edf9b324465d14b2efca169a25d564a0.png?width=2000&#x26;format=png&#x26;optimize=medium" media="(min-width: 600px)">
-                <img loading="lazy" alt="Hero image" type="image/png" src="./media_a22b1a53edf9b324465d14b2efca169a25d564a0.png?width=750&#x26;format=png&#x26;optimize=medium">
+                <img loading="lazy" alt="Hero image" data-title="" type="image/png" src="./media_a22b1a53edf9b324465d14b2efca169a25d564a0.png?width=750&#x26;format=png&#x26;optimize=medium">
```
[the remark this answers is not in the document]
Comment (tmaret) [reply test/fixtures/content/image-from-meta.html]: @tripodsan I have pushed the change. It turns out that hastscript outputs empty string properties and skips undefined ones.

File: src/steps/create-pictures.js [hunk 2]
```diff
@@ -55,6 +55,7 @@ export function createOptimizedPicture(src, alt = '') {
     return h('img', {
       loading: 'lazy',
       alt,
+      'data-title': title === alt ? '' : title,
```
Comment (tripodsan) [inline src/steps/create-pictures.js:58]:
```suggestion
      'data-title': title === alt ? undefined : title,
```

**Review by tripodsan — APPROVED** [review]

**Pull request merged**